packages/insomnia/src/ui/components/viewers/response-csv-viewer.tsx

```diff
@@ -1,73 +1,40 @@
-import { autoBindMethodsForReact } from 'class-autobind-decorator';
 import Papa from 'papaparse';
-import React, { PureComponent } from 'react';
-
-import { AUTOBIND_CFG } from '../../../common/constants';
+import React, { FC, useCallback, useEffect, useState } from 'react';
 
 interface Props {
   body: Buffer;
 }
 
-interface State {
-  result: null | {
-    data: string[][];
-  };
-}
-
-@autoBindMethodsForReact(AUTOBIND_CFG)
-export class ResponseCSVViewer extends PureComponent<Props, State> {
-  state: State = {
-    result: null,
-  };
-
-  currentHash = '';
-
-  update(body: Buffer) {
-    const csv = body.toString('utf8');
-    Papa.parse<string[]>(csv, {
+export const ResponseCSVViewer: FC<Props> = ({ body }) => {
+  const [csv, setCSV] = useState<{ data: string[][] } | null>(null);
+  const parse = useCallback(() => {
```

Comment:
This could be changed to useEffect since it's not used in an event handler.

```suggestion
const parse = useEffect(() => {
```